apps/earn-protocol/helpers/chart-helpers/get-ark-historical-data.ts

```
@@ -0,0 +1,263 @@
+      ...acc,
+      [key]: data.sort((a, b) => dayjs(a.timestamp).unix() - dayjs(b.timestamp).unix()), // final sorting
+    } as ChartsDataTimeframes
+  }, emptyChart)
+
+  const chartColors = Object.keys(arkInterestRatesMap).reduce((acc, key) => {
+    // temporary mapping, we need something more robust from subgraph
+    const protocol = key.split('-')
+    const arkUniqueName = getProtocolLabel(protocol)
+
+    return {
+      ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)